esp/platform/txsummary.cpp

```diff
@@ -65,100 +398,490 @@ bool CTxSummary::contains(const char* key) const
     return __contains(key);
 }
 
-bool CTxSummary::append(const char* key, const char* value, const LogLevel logLevel)
+bool CTxSummary::append(const char* key, const char* value, const LogLevel logLevel, const unsigned int group)
+{
+    if(!validateKey(key))
+        return false;
+
+    CriticalBlock block(m_sync);
+
+    if (__contains(key))
+        return false;
+
+    EntryValue entry;
+    entry.setown(new TxEntryStr(key, value, logLevel, group));
```

Comment:
Create the new entry before `block`. There's an opportunity to refactor the multiple overloads to act on the entry to be added.